app/controllers/admin/volunteers.py

```diff
@@ -81,16 +84,32 @@ def updateVolunteerTable(eventID):
         flash("Error adding volunteer", "danger")
     return redirect(url_for("admin.trackVolunteersPage", eventID=eventID))
 
-@admin_bp.route('/addVolunteerToEvent/<volunteer>/<eventId>', methods = ['POST'])
-def addVolunteer(volunteer, eventId):
-    username = volunteer.strip("()").split('(')[-1]
+@admin_bp.route('/addVolunteerToEvent', methods = ['POST'])
+def addVolunteer():
+    volunteerData = request.form
+    username = volunteerData["username"]
     user = User.get(User.username==username)
+    eventId = volunteerData['eventId'][0]
     successfullyAddedVolunteer = addVolunteerToEventRsvp(user, eventId)
     EventParticipant.create(user=user, event=eventId) # user is present
     if successfullyAddedVolunteer:
         flash("Volunteer successfully added!", "success")
     else:
-        flash("Error when adding volunteer", "danger")
+        flash("Error when adding vol    unteer", "danger")
+    return ""
+
+@admin_bp.route('/addOutsideParticipantToEvent', methods = ['POST'])
+def addOutsideParticipant():
+
+    outsideParticipantData = request.form
+    email = outsideParticipantData['email']
+    eventId = outsideParticipantData['eventId']
+    event = eventId.split(':')
```

Comment:
I would do the whole calculation here. Because what is being stored in `event` ISN'T an event, it is an array. Also, why are you passing something that isn't an event ID already in a field called `eventId`?